pkg/manager/manager.go

```diff
@@ -10,31 +10,54 @@ import (
 	managercfg "github.com/kong/kubernetes-ingress-controller/v3/pkg/manager/config"
 )
 
-// Manager is an object representing an instance of the Kong Ingress Controller.
-type Manager struct {
-	id     ID
-	config managercfg.Config
-	logger logr.Logger
-}
-
 // NewManager creates a new instance of the Kong Ingress Controller. It does not start the controller.
-func NewManager(id ID, logger logr.Logger, configOpts ...managercfg.Opt) (*Manager, error) {
+func NewManager(ctx context.Context, id ID, logger logr.Logger, configOpts ...managercfg.Opt) (*Manager, error) {
 	cfg, err := NewConfig(configOpts...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create manager config: %w", err)
 	}
 
+	m, err := managerinternal.New(ctx, cfg, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create manager: %w", err)
+	}
+
 	return &Manager{
-		id:     id,
-		config: cfg,
-		logger: logger.WithValues("managerID", id.String()),
+		id:      id,
+		config:  cfg,
+		logger:  logger.WithValues("managerID", id.String()),
+		manager: m,
 	}, nil
 }
 
+// Manager is an object representing an instance of the Kong Ingress Controller.
+type Manager struct {
+	id      ID
+	config  managercfg.Config
+	logger  logr.Logger
+	manager *managerinternal.Manager
+}
```

Comment:
Can we move the struct definition above the constructor? I believe that's the convention we most often follow.